lib/using_restricted_array.rb

```diff
@@ -1,61 +1,118 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear or (O)n, where n is the number of items in
+# the array or the length of the input array
+# Space complexity: constant O(1)
 def length(array)
-  raise NotImplementedError
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear O(n), where n is the lenght of the array
+# Space complexity: constant O(1)
+
 def print_array(array)
-  raise NotImplementedError
+  len = length(array)
```

Comment:
[nit] You can technically save yourself the additional time to run length(array) by using until array[i] == nil in this method the way you did for length. Normally, when using built-in arrays, I would prefer what you have here because it's fast to get the length of a normal array. In this case, though, not using length(array) would save actual time (though not big-O time).